argusAI/event_similarity_processor_v2.py

```diff
@@ -0,0 +1,234 @@
+                if batch_processed == 0:
+                    # No events to process, sleep before next iteration
+                    time.sleep(SLEEP_INTERVAL)
+                else:
+                    LOGGER.info(f"Processed {batch_processed} events in this batch")
+            except Exception as e:
+                LOGGER.error(f"Error in processing loop: {e}", exc_info=True)
+                self.error_count += 1
+                time.sleep(SLEEP_INTERVAL)
+
+        LOGGER.info(f"Processing loop stopped. Total processed: {self.processed_count}, Errors: {self.error_count}")
+
+    def _process_batch(self, batch_size: int = 100) -> int:
+        """
+        Process a batch of unprocessed events.
+
+        Args:
+            batch_size: Maximum number of events to process in one batch
+
+        Returns:
+            Number of events processed in this batch
+        """
+        # Fetch unprocessed events using raw query
+        query = f"SELECT run_id, severity, ts FROM sct_unprocessed_events LIMIT {batch_size}"
+        try:
+            rows = self.db.execute(query)
+            unprocessed_events = list(rows)
+        except Exception as e:
+            LOGGER.error(f"Failed to fetch unprocessed events: {e}", exc_info=True)
+            return 0
+
+        if not unprocessed_events:
+            return 0
+
+        processed_in_batch = 0
+
+        for event_row in unprocessed_events:
+            try:
+                self._process_single_event(run_id=event_row.run_id, severity=event_row.severity, ts=event_row.ts)
+                processed_in_batch += 1
+                self.processed_count += 1
+            except Exception as e:
+                LOGGER.error(
+                    f"Failed to process event (run_id={event_row.run_id}, "
+                    f"severity={event_row.severity}, ts={event_row.ts}): {e}",
+                    exc_info=True,
+                )
+                self.error_count += 1
+                # Still delete the unprocessed event to avoid infinite retries
+                delete_query = "DELETE FROM sct_unprocessed_events WHERE run_id = ? AND severity = ? AND ts = ?"
+                self.db.execute(delete_query, (event_row.run_id, event_row.severity, event_row.ts))
+
+        return processed_in_batch
+
+    def _process_single_event(self, run_id: UUID, severity: str, ts: datetime) -> None:
+        """
+        Process a single event: read, sanitize, embed, store, and cleanup.
+
+        Args:
+            run_id: Test run UUID
+            severity: Event severity (ERROR or CRITICAL)
+            ts: Event timestamp
+
+        Raises:
+            Exception: If event not found or embedding generation fails
+        """
+        LOGGER.debug(f"Processing event: run_id={run_id}, severity={severity}, ts={ts}")
+
+        # Step 1: Read event message from SCTEvent table
+        try:
+            query = "SELECT message FROM sct_event WHERE run_id = ? AND severity = ? AND ts = ?"
```

Comment:
Ditto

Reply:
Here actually cqlengine could be used provided it has been initialized - the query is very simple.